### PR TITLE
REF: implement Index._get_indexer_strict

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

xref #27181, a couple more steps to get that all the way accomplished

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -5390,6 +5390,89 @@ def get_indexer_for(self, target) -> np.ndarray:
...
+
+        if self._index_as_unique:
+            indexer = self.get_indexer_for(keyarr)
+            keyarr = self.reindex(keyarr)[0]
+        else:
+            keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
+
+        self._raise_if_missing(keyarr, indexer, axis_name)
+
+        if (
```
Comment (jreback) [inline pandas/core/indexes/base.py:5409]: is_extension_array_dtype?

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -5390,6 +5390,89 @@ def get_indexer_for(self, target) -> np.ndarray:
...
+        if len(key) == 0:
+            return
+
+        # Count missing values
+        missing_mask = indexer < 0
+        nmissing = missing_mask.sum()
+
+        if nmissing:
+
+            # TODO: remove special-case; this is just to keep exception
```
Comment (jreback) [inline pandas/core/indexes/base.py:5458]: what does this mean?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:5458]: when we removed IntervalIndex._convert_listlike_indexer we put special-casing here to keep the exception message unchanged.  This comment is to remove that special-casing.

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:5409]: also dt64 and td64.  im pretty sure we should actually be doing this (.take on L5416) unconditionally, but that breaks a few tests which i think are wrong, so thats not ready yet

**Comment by jbrockmendel** [comment]
rebased + greenish

**Comment by jbrockmendel** [comment]
@jreback gentle ping, this unblocks the last 2 MultiIndex PRs on deck

**Review by jreback — APPROVED** [review]

**Pull request merged**